src/dev/performance/ingest_metrics.ts

```diff
@@ -0,0 +1,92 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CiStatsReporter, ToolingLog } from '@kbn/dev-utils';
+import { LoadingFinishedEvent, ResponseReceivedEvent } from './event';
+
+export function ingestPerformanceMetrics(
+  log: ToolingLog,
+  responses: Map<
+    string,
+    Map<string, { loadingFinished: LoadingFinishedEvent; responseRecieved: ResponseReceivedEvent }>
+  >
+) {
+  const uniqueRequests = new Map<
+    string,
+    Array<{ path: string; encodedBodyLength: number; plugin?: string; type?: string }>
+  >();
+
+  for (const [path, frameResponses] of responses) {
+    for (const [, response] of frameResponses) {
+      const url: string = response.responseRecieved.response.url;
+      const encodedBodyLength: number =
+        response.loadingFinished.encodedDataLength -
+        response.responseRecieved.response.encodedDataLength;
+      if (uniqueRequests.has(url)) {
+        uniqueRequests.set(
+          url,
+          [{ path, encodedBodyLength }].concat(uniqueRequests.get(url) || [])
+        );
+      } else {
+        uniqueRequests.set(url, [{ path, encodedBodyLength }]);
+      }
+    }
+  }
+
+  const metrics = Array<{
+    url: string;
+    encodedBodyLength: number;
+    plugin?: string;
+    type?: string;
+  }>();
+
+  for (const [url, array] of uniqueRequests) {
+    const isPlugin = url.endsWith('.plugin.js');
+    if (isPlugin) {
+      const type = url.match(/.*\/\d+.plugin.js$/) ? 'lazy' : 'entry';
+      const results = url.match(/(?<=plugin\/)(.*)(?=\/)/);
+      const plugin = results ? results[0] : url;
+      const encodedBodyLength = Math.round(
+        array.map((i: any) => i.encodedBodyLength).reduce((acc, value) => acc + value) /
+          array.length
+      );
+
+      metrics.push({ url, plugin, type, encodedBodyLength });
+    } else {
+      if (array.length > 1) {
+        const encodedBodyLength = Math.round(
+          array.map((i: any) => i.encodedBodyLength).reduce((acc, value) => acc + value) /
+            array.length
+        );
+        metrics.push({ url, encodedBodyLength });
+      }
+    }
+  }
+
+  const reporter = CiStatsReporter.fromEnv(log);
+  metrics.map(asset => {
+    reporter.metrics([
+      {
+        group: 'Asset size',
+        id: asset.url,
+        value: asset.encodedBodyLength,
+      },
+    ]);
+  });
+}
```

Comment:
I talked to @restrry and maybe we want to report not only plugin entry assets, but also the ones that are loaded lazy.

I'm not sure how to put it nicely with current `reporter.metrics` implementation.

@restrry suggested smth like:

```
{
        group: 'Asset size',
        id: asset.path,
        type: 'entry' | 'lazy',
        plugin: pluginName,
        value: asset.encodedBodyLength,
      },
```

Reply:
I think that reporting on "lazy" bundles will make a lot more noise than we want personally. Between module count and page load size I think we have plenty of information, and I also think if we wanted to get the size of all files it would make more sense to just get the asset size from the filesystem.

Reply:
> I think that reporting on "lazy" bundles will make a lot more noise than we want personally

On the one hand, it's important to track that lazy loaded chunks aren't bloated. OTOH it's hard to predict when and under what circumstances they will be loaded. I think we are good to use the whole Kiban artifact size instead.

> I also think if we wanted to get the size of all files it would make more sense to just get the asset size from the filesystem.

Why didn't we go this way from the very beginning?

Reply:
Primarily because I think the assets which are loaded on first page load are an important metric, and I suspect this is the easiest way (without completely reachitecting how we load assets) to determine which assets are loaded on every page.

The goal was to determine how large the overhead of all applications is, the entry bundles that are produced by each plugin and loaded in every application whether the app uses that logic or not, with the intention of leading more devs to load code on demand, when it's needed.